lib/blockscore/collection.rb

```diff
@@ -1,18 +1,91 @@
 module BlockScore
   class Collection < Array
-    def initialize(target)
-      super()
-      @target = target
+    attr_reader :default_params
+    protected :default_params
+
+    attr_reader :parent, :target, :data
+
+    def initialize(params)
+      @parent = params.fetch :parent
+      @target = params.fetch :target
+      register_parent_data
+    end
+
+    def default_params
+      {
+        :"#{parent_name}_id" => parent_id
+      }
+    end
+
+    def parent_id
+      parent.id if parent.respond_to?(:id)
```

Comment:
Hm ok.  What do you think about just defining an explicit id method in the resource base class?  Something simple like

```ruby
def id
  attributes.fetch(:id, nil)
end
```

so that classes like `Collection` don't have to introspect on resource classes like this